src/Symfony/Component/Validator/Constraints/EnumValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * EnumValidator validates that the value is one of the expected values.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class EnumValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Enum) {
+            throw new UnexpectedTypeException($constraint, Enum::class);
+        }
+
+        if (!enum_exists($constraint->type) || !method_exists($constraint->type, 'from')) {
```

Comment:
Unnecessary duck-typing.

```suggestion
        if (!is_a($constraint->type, \BackedEnum::class, true)) {
```